src/image/ImageManagerActions.cc

```diff
@@ -560,8 +593,25 @@ int ImageManager::delete_image(int iid, string& error_str)
 
         if ( rc < 0 )
         {
-            NebulaLog::log("ImM",Log::ERROR,
-                "Image could not be removed from DB");
+            NebulaLog::log("ImM", Log::ERROR,
+                           "Image could not be removed from DB");
+
+            return -1;
+        }
+
+        int uid = img->get_uid();
+        int gid = img->get_gid();
+        long long size = img->get_size() + img->get_snapshots().total_size();
+
+        img.reset();
+
+        Quotas::ds_del(uid, gid, ds_id, size);
+
+        // Remove Image reference from Datastore
+        if ( auto ds = dspool->get(ds_id) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator